core/templates/pages/learner-dashboard-page/new-learner-dashboard/progress-tab.component.ts

```diff
@@ -31,15 +32,65 @@ import { LearnerTopicSummary } from 'domain/topic/learner-topic-summary.model';
 export class ProgressTabComponent implements OnInit {
   @Input() completedStoriesList: StorySummary[];
   @Input() partiallyLearntTopicsList: LearnerTopicSummary[];
+  @Input() learntTopicsList: LearnerTopicSummary[];
+  topicsInSkillProficiency: LearnerTopicSummary[] = [];
+  emptySkillProficiency: boolean = true;
+  displaySkills: boolean[];
   widthConst: number = 233;
+  subtopicMastery: Record<string, SubtopicMasterySummaryBackendDict> = {};
+  topicIdsInSkillProficiency: string[] = [];
+  goldBadgeImageUrl: string = '';
+  bronzeBadgeImageUrl: string = '';
+  silverBadgeImageUrl: string = '';
+  emptyBadgeImageUrl: string = '';
+  topicMastery: number[] = [];
   width: number;
 
   constructor(
     private deviceInfoService: DeviceInfoService,
     private urlInterpolationService: UrlInterpolationService,
+    private learnerDashboardBackendApiService: LearnerDashboardBackendApiService
   ) {}
 
-  ngOnInit(): void {
+  async ngOnInit(): Promise<void> {
+    this.width = this.widthConst * (this.completedStoriesList.length);
+    this.topicsInSkillProficiency.push(
+      ...this.partiallyLearntTopicsList, ...this.learntTopicsList);
+    let topic: LearnerTopicSummary;
+    for (topic of this.topicsInSkillProficiency) {
+      this.topicIdsInSkillProficiency.push(topic.id);
+    }
+    this.goldBadgeImageUrl = this.getStaticImageUrl(
+      '/learner_dashboard/gold.png');
+    this.bronzeBadgeImageUrl = this.getStaticImageUrl(
+      '/learner_dashboard/bronze.png');
+    this.silverBadgeImageUrl = this.getStaticImageUrl(
+      '/learner_dashboard/silver.png');
+    this.emptyBadgeImageUrl = this.getStaticImageUrl(
+      '/learner_dashboard/empty_badge.png');
+    if (this.topicsInSkillProficiency.length !== 0) {
+      this.subtopicMastery = await (
+        this.learnerDashboardBackendApiService.getFetchSubtopicMastery(
+          this.topicIdsInSkillProficiency.join(',')));
+    }
+    this.displaySkills = new Array(
+      this.topicsInSkillProficiency.length).fill(false);
+    let atLeastOnetopicHasPracticeTabEnabled = false;
+    for (topic of this.topicsInSkillProficiency) {
+      if (topic.practiceTabIsDisplayed === true) {
```

Comment:
What is this field supposed to mean? Is it that it should be displayed in practice tab? If so, rename it to `topic.displayedInPracticeTab`

Reply:
This field means that the practice tab is enabled in the topic. I think it should be `practiceTabIsDisplayed `. What do you think?

Reply:
Oh ok, yeah, it's fine then.